packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -1717,23 +1754,29 @@ class _SelectableFragment with Selectable, ChangeNotifier implements TextLayoutM
   }
 
   SelectionResult _handleSelectWord(Offset globalPosition) {
-    _selectableContainsOriginWord = true;
+    final Matrix4 transform = paragraph.getTransformTo(null);
+    transform.invert();
+    final Offset localPosition = MatrixUtils.transformPoint(transform, globalPosition);
+    if (!_rect.contains(localPosition) && (isFollowedByInlineElement || isInlineWidget)) {
+      return SelectionResult.forward;
+    }
 
     final TextPosition position = paragraph.getPositionForOffset(paragraph.globalToLocal(globalPosition));
     if (_positionIsWithinCurrentSelection(position) && _textSelectionStart != _textSelectionEnd) {
       return SelectionResult.end;
     }
     final _WordBoundaryRecord wordBoundary = _getWordBoundaryAtPosition(position);
-    if (wordBoundary.wordStart.offset < range.start && wordBoundary.wordEnd.offset < range.start) {
+    if (wordBoundary.wordStart.offset < range.start && wordBoundary.wordEnd.offset <= range.start) {
```

Comment:
This change was made to handle the case when `SelectWordSelectionEvent` is dispatched with a position that is on a placeholder in the text. In this case `_getWordBoundaryAtPosition` may return the range of the placeholder which touches the edges of the selectable fragments range.